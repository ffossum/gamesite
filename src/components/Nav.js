import React, { PropTypes } from 'react';
import { Link } from 'react-router';
import Gravatar from 'components/common/Gravatar';

import styles from './nav.css';

export default class Nav extends React.Component {
  render() {
    return (
      <nav role="navigation" className={styles.navbar}>
        <ul className={styles['nav-item']}>
          <li><Link to="/">Main</Link></li>
          <li><Link to="/play">Play</Link></li>
          <li><Link to="/about">About</Link></li>
        </ul>

        {
          this.props.user
          ?
          <ul>
            <li className={styles.user}>
              <Gravatar inline emailHash={this.props.user.emailHash} />
              {this.props.user.username}
            </li>
            <li className={styles['nav-item']}>
              <a href="#" onClick={this.props.logOut}>Log out</a>
            </li>
          </ul>
          :
          <ul className={styles['nav-item']}>
            <li><Link to="/login">Log in</Link></li>
            <li><Link to="/register">Register</Link></li>
          </ul>
        }
      </nav>
    );
  }
}

Nav.propTypes = {
  user: PropTypes.shape({
    username: PropTypes.string.isRequired,
    emailHash: PropTypes.string.isRequired,
  }),
  logOut: PropTypes.func.isRequired,
};
