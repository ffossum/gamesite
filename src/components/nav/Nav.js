import React, { PropTypes } from 'react';
import { IndexLink, Link } from 'react-router';
import GamesDropdown from './GamesDropdown';
import UserDropdown from './UserDropdown';
import _ from 'lodash';
import {
  LOGIN_MODAL,
  REGISTER_MODAL,
} from 'constants/modalType';

import styles from './nav.css';

export default class Nav extends React.Component {
  constructor() {
    super();

    this.openLoginModal = this.openModal.bind(this, LOGIN_MODAL);
    this.openRegisterModal = this.openModal.bind(this, REGISTER_MODAL);
  }
  openModal(modalType, e) {
    e.preventDefault();
    this.props.openModal(modalType);
  }
  render() {
    const { user, games } = this.props;

    return (
      <nav role="navigation" className={styles.navbar}>
        <ul className={styles.navgroup}>
          <li>
            <IndexLink
              to="/"
              className={styles.navlink}
            >Main</IndexLink>
          </li>
          <li>
            <Link
              to="/play"
              activeClassName={styles.active}
              className={styles.navlink}
            >Play</Link>
          </li>
          <li>
            <Link
              to="/about"
              activeClassName={styles.active}
              className={styles.navlink}
            >About</Link>
          </li>
        </ul>

        {
          user ?
            <ul className={styles.navgroup}>
              <li><GamesDropdown games={games} /></li>
              <li><UserDropdown user={user} logOut={this.props.logOut} /></li>
            </ul>
          :
            <ul className={styles.navgroup}>
              <li>
                <a className={styles.navlink} href="" onClick={this.openLoginModal}>Log in</a>
              </li>
              <li>
                <a className={styles.navlink} href="" onClick={this.openRegisterModal}>Register</a>
              </li>
            </ul>
        }
      </nav>
    );
  }
}

Nav.propTypes = {
  user: PropTypes.shape({
    username: PropTypes.string.isRequired,
    emailHash: PropTypes.string.isRequired,
  }),
  games: PropTypes.object,
  logOut: PropTypes.func.isRequired,
  openModal: PropTypes.func.isRequired,
};
