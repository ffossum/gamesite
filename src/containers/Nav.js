import React, {PropTypes} from 'react';
import {Link} from 'react-router';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import actions from 'actions/login';

import styles from './nav.css';

export default class Nav extends React.Component {
  render() {
    return (
      <nav role="navigation" className={styles.navbar}>
        <div className={styles['nav-item']}>
          <Link to="/">Main</Link>
        </div>

        {
          this.props.loggedInUser
          ?
          <ul>
            <li className={styles.user}>
              User id: {this.props.loggedInUser}
            </li>
            <li className={styles['nav-item']}>
              <a href="#" onClick={this.props.logOut}>Log out</a>
            </li>
          </ul>
          :
          <ul className={styles['nav-item']}>
            <li><Link to="/login">Log in</Link></li>
            <li><Link to="/register">Register</Link></li>
          </ul>
        }
      </nav>
    );
  }
}

Nav.propTypes = {
  loggedInUser: PropTypes.string,
  logOut: PropTypes.func.isRequired
};

export default connect(
  state => ({
    loggedInUser: state.loggedInUser
  }),
  dispatch => bindActionCreators(actions, dispatch)
)(Nav);
