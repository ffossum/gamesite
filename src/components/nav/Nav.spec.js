/* @flow */
/* eslint-env jest */

import React from 'react';
import Nav from './Nav';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const logOut = () => {};
const openModal = () => {};

describe('Nav', () => {
  test('renders correctly', () => {
    const component = shallow(<Nav logOut={logOut} openModal={openModal} />);
    expect(toJSON(component)).toMatchSnapshot();
  });
});
