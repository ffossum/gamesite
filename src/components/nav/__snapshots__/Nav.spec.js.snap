exports[`Nav renders correctly 1`] = `
<nav
  className="navbar">
  <ul
    className="navgroup">
    <li>
      <IndexLink
        className="navlink"
        to="/">
        Main
      </IndexLink>
    </li>
    <li>
      <Link
        activeClassName="active"
        className="navlink"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/play">
        Play
      </Link>
    </li>
    <li>
      <Link
        activeClassName="active"
        className="navlink"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/about">
        About
      </Link>
    </li>
  </ul>
  <ul
    className="navgroup">
    <li>
      <a
        className="navlink"
        href=""
        onClick={[Function]}>
        Log in
      </a>
    </li>
    <li>
      <a
        className="navlink"
        href=""
        onClick={[Function]}>
        Register
      </a>
    </li>
  </ul>
</nav>
`;
